src/compiler/checker.ts

```diff
@@ -5451,6 +5453,11 @@ namespace ts {
                                 localErrors = false;
                             }
                         }
+                        if (reportErrors) {
+                            reportError(Diagnostics.Signature_0_has_no_corresponding_signature_in_1,
+                                signatureToString(t, /*enclosingDeclaration*/ undefined, /*flags*/ undefined, kind),
+                                typeToString(source));

```

Comment:
I think this error message has it backwards. The source type is required to provide a matching signature for every signature in the target type. So, the message should be `Type '{0}' provides no match for the signature '{1}'` and then flip the two arguments.